### ISSUE TITLE
Lint/SymbolConversion false positive for Hash keys that contain ":"

### ISSUE THREAD

**Issue by s-mage** [rubocop-hq/rubocop#9455]
Hello. Thanks for your work on the rubocop, it's a great tool.

I've found a false positive for `Lint/SymbolConversion` cop. It reacts on hash keys like `:"one:two"` and suggests to replace them with `:onetwo`. It doesn't fail on `var = :"one:two"` though.

--------

## Expected behavior

Don't get triggered on hashes with keys that contain `:`.

## Actual behavior

```
{ "one:two": "three" }
# [Lint/SymbolConversion] [W]  Unnecessary symbol conversion; use `"onetwo":` instead.
```

## Steps to reproduce the problem

Any hash with key like `:"something:something_else"` shows the warning.

## RuboCop version

```
->  rubocop -V
warning: parser/current is loading parser/ruby26, which recognizes
warning: 2.6.6-compliant syntax, but you are running 2.6.5.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
1.9.0 (using Parser 3.0.0.0, rubocop-ast 1.4.1, running on ruby 2.6.5 x86_64-darwin19)
  - rubocop-performance 1.9.2
  - rubocop-rails 2.9.1
```

**Comment by s-mage** [comment]
Just found another one: 

```
{ "one=": 1 } # [Lint/SymbolConversion] [W]  Unnecessary symbol conversion; use `one=:` instead. ...
```

```
> { one=: 1 }
SyntaxError: unexpected ':'
{ one=: 1 }
```

Not sure if it should be a separate issue

**Comment by koic** [comment]
@s-mage Yeah, maybe a slightly different issue. Can you please open a new another issue?
